### ISSUE TITLE
outExtension option appears to produce ambiguity

### ISSUE THREAD

**Issue by verikono** [evanw/esbuild#3028]
I have a project which will output multiple files and contain .mts files and also contain .tsx files. I need the .mts to target extension .mjs and .tsx files to target extension .js. 

the outExtension options seems like the goto but it's design specifies the key of the option object be  { [emitted extension]: [desired extension] } yet both .tsx and .mts files have been emitted as .js files and esbuild can't distinguish the two outputs....

might check to see if there's a plugin for this, or maybe I'll just whack one together.... or maybe i'm missing something?

**Comment by Inqnuam** [comment]
Have you tried this? :

```js
import * as esbuild from "esbuild";

await esbuild.build({
  entryPoints: [
    { in: "src/home.tsx", out: "[dir]/[name].js" },
    { in: "src/utils.mts", out: "[dir]/[name].mjs" },
  ],
  bundle: true,
  write: true,
  outdir: "dist",
});

```

**Comment by evanw** [comment]
Yes, this is not currently able to be specified. You can still get this to work by calling `esbuild.build` multiple times with different options for different entry points.

**Comment by verikono** [comment]
Thanks for your reply guys :)

I actually ended up using this as an excuse to write a plugin - maybe i
should publicise it even though both your reponses are completely
workable and good from the prom :)
